src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,650 @@
+r"""
+Ordered Hyperplane Arrangements
+
+The :class:`HyperplaneArrangements` ordered the arrangements in a hyperplane
+independently of the way the hyperplanes are introduced. With the class
+:class:`OrderedHyperplaneArrangements` it is allowed to fix an order by
+the user, what is needed for some properties, e.g., topological one.
+Besides this fact, it inherits the rest of the properties from :class:`HyperplaneArrangements.`
+
+Ordered arrangements
+--------------------
+
+An ordered arrangement is an arrangement where the hyperplanes are sorted
+by the user::
+
+    sage: H0.<t0, t1, t2> = HyperplaneArrangements(QQ)
+    sage: H0(t0 - t1, t1 - t2, t0 - t2)
+    Arrangement <t1 - t2 | t0 - t1 | t0 - t2>
+    sage: H.<t0, t1, t2> = OrderedHyperplaneArrangements(QQ)
+    sage: H(t0 - t1, t1 - t2, t0 - t2)
+    Arrangement <t0 - t1 | t1 - t2 | t0 - t2>
+
+Some methods are adapted and some new ones are created, regarding
+hyperplane sections and fundamental groups, e.g., ``hyperplanes``::
```

Comment:
```suggestion
hyperplane sections and fundamental groups, e.g., :meth:`~XYZ.hyperplanes`::
```
where `XYZ` is the appropriate class (with the file path). Similar comment applies below.

Reply:
I missed this one, it will be done in the following commit.